File: beecrowd/exercicio5.c
#include <stdio.h>

int main(){
    int n;

    while(scanf("%d", &n) != EOF){
        long long tempoCpuLivre = 0;
        long long tempoTotalEspera = 0;

        for(int i = 0; i < n; i++){
            int t,c;
            scanf("%d %d", &t, &c);

            long long tempoInicioExecucao;
            if(t > tempoCpuLivre){
                tempoInicioExecucao = t;
            } else{
                tempoInicioExecucao = tempoCpuLivre;
            }

            long long tempoEspera = tempoInicioExecucao - t;
            tempoTotalEspera += tempoEspera;

            tempoCpuLivre = tempoInicioExecucao + c;
        }
        printf("%d\n", tempoTotalEspera);
    }
    return 0;
}

File: beecrowd/exercicio5.java
import java.util.Scanner;

public class exercicio5 {
    
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);

        while(sc.hasNext()){
            int n = sc.nextInt();
            long tempoCpuLivre = 0;
            long tempoTotalEspera = 0;

            for(int i = 0; i < n; i++){
                int t = sc.nextInt();
                int c = sc.nextInt();

                long tempoInicioExecucao;
                
                if(t > tempoCpuLivre){
                    tempoInicioExecucao = c;
                } else{
                    tempoInicioExecucao = tempoCpuLivre;
                }
                long tempoEspera = tempoInicioExecucao - t;
                tempoTotalEspera += tempoEspera;

                tempoCpuLivre = tempoInicioExecucao + c;
            }
            System.out.println(tempoTotalEspera);
        }
        sc.close();
    }
}
